pkg/operator/observe_config.go

```diff
@@ -97,6 +103,60 @@ func (c ConfigObserver) sync() error {
 	return nil
 }
 
+// observeClusterConfig observes cloud provider configuration from
+// cluster-config-v1 in order to configure kube-controller-manager's cloud
+// provider.
+func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	clusterConfig, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
+		return observedConfig, nil
+	}
+	if err != nil {
+		return observedConfig, err
```

Comment:
nit: in the future, might be good to just explicitly return `nil` here for observedConfig